### DISCUSSION TITLE
How do you create a ssh key with write access?

### DISCUSSION THREAD

**Discussion by carlosjgp** [kelvintaywl/terraform-provider-circleci#51] [Q&A] [answered]
This :point_down: on "project > SSH keys" section at the bottom

![image](https://github.com/kelvintaywl/terraform-provider-circleci/assets/1277613/773b96fe-634d-4ce1-9d1a-1da18ce2d8c3)

**Comment by kelvintaywl** [comment, marked as answer]
Hi @carlosjgp 

sorry for the late follow-up.
I had just come back from parental leave. 👶 

were you able to resolve this?

otherwise, please allow me to try to address your question 😄 

There are 2 types of SSH keys you can create for a CircleCI project:

1. a checkout key (used to git clone your repo)
2. [a general SSH key](https://circleci.com/docs/add-ssh-key/#overview) (useful for your CircleCI job to SSH into a remote server like EC2 instance for example).

The screenshot you are sharing is for (2).
Right now, there is no way to create a SSH key for a CircleCI project via this provider.
This is because there is no public APIs from CircleCI to add a SSH key.
_You can do so manually via the UI on CircleCI still._

_If_ you are referring to a checkout key where you can also **push (write) to your repo**, then you are referring to (1).
For this, you need [a user key (not deploy key)](https://circleci.com/docs/github-integration/#deploy-keys-and-user-keys).
You can then use this provider to generate a user key (via the `circleci_checkout_key` resource).
See example here:
https://registry.terraform.io/providers/kelvintaywl/circleci/latest/docs/resources/checkout_key#creating-user-keys
